.github/workflows/go-tests.yml

```diff
@@ -0,0 +1,15 @@
+on: [push, pull_request]
+name: Test
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.20.x]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```

Comment:
Why is it necessary to check the tests on different go versions and different OSs? [CometBFT](https://github.com/cometbft/cometbft/blob/main/.github/workflows/tests.yml), [SDK](https://github.com/cosmos/cosmos-sdk/blob/main/.github/workflows/test.yml#L113), and [IBC](https://github.com/cosmos/ibc-go/blob/main/.github/workflows/test.yml) are all checking only for ubuntu-latest and the go version in go.mod.

Reply:
because people may want to run chains on windows or mac